pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java

```diff
@@ -101,10 +112,33 @@ protected InstanceResponseBlock getNextBlock() {
       responseMetaData
           .put(MetadataKey.SYSTEM_ACTIVITIES_CPU_TIME_NS.getName(), String.valueOf(systemActivitiesCpuTimeNs));
 
+      // TODO: Remove this once the SelectionOrderByOperator is modified to throw an exception to catch cases where
+      //       an MV column (identifier or via transform) is present on the order-by list
+      logAndEmitMetricForQueryHasMVSelectionOrderBy(intermediateResultsBlock);
+
       return instanceResponseBlock;
     } else {
-      return new InstanceResponseBlock(getCombinedResults());
+      // TODO: Remove this once the SelectionOrderByOperator is modified to throw an exception to catch cases where
+      //       an MV column (identifier or via transform) is present on the order-by list
+      IntermediateResultsBlock intermediateResultsBlock = getCombinedResults();
+      logAndEmitMetricForQueryHasMVSelectionOrderBy(intermediateResultsBlock);
```

Comment:
Suggest doing a bit of cleanup

- Move `IntermediateResultBlock` and `InstanceResponseBlock` declaration before if-else block
- Call logA`ndEmitMetricForQueryHasMVSelectionOrderBy(intermediateResultsBlock)` outside if-else
- return `instanceResponseBlock` outside if-else